FirebaseTestProject/Models/Post.swift

```diff
@@ -49,13 +79,37 @@ struct PostService {
         return db.collection("posts")
     }
     
+    static var favoritesReference: CollectionReference {
+        let db = Firestore.firestore()
+        return db.collection("favorites")
+    }
+    
     static func getPosts() async throws -> [Post] {
         let postsSnapshots = try await postsReference.getDocuments()
         let posts = postsSnapshots.documents.map { Post(from: $0.data()) }
         return posts
     }
     
+    static func getFavorites() async throws -> [Post] {
+        let userid = UserDefaults.standard.value(forKey: "userid") != nil ?  UserDefaults.standard.value(forKey: "userid") as! String : "00854E9E-8468-421D-8AA2-605D8E6C61D9"
+        let favoritesQuery = favoritesReference.whereField("userid", isEqualTo: userid)
+        let favoritesSnapshots = try await favoritesQuery.getDocuments()
+        let favorites = favoritesSnapshots.documents.map { Post(from: $0.data()) }
+        
+        return favorites
+    }
+    
     static func upload(_ post: Post) async throws {
         try await postsReference.document(post.id.uuidString).setData(post.jsonDict)
     }
+    
+    static func favorite(_ post: Post) async throws {
+        let userid = UserDefaults.standard.value(forKey: "userid") as? String ?? "00854E9E-8468-421D-8AA2-605D8E6C61D9"
+        try await favoritesReference.document(post.id.uuidString).setData(Favorite(postid: post.id.uuidString, userid: userid).jsonDict)
+    }
+    
+    // Added for the Delete functionality in the other branch
+    static func delete(favorite: Post) async throws {
+        try await favoritesReference.document(favorite.id.uuidString).delete()
```

Comment:
In addition to the notes above regarding primary keys and user IDs, I’m wondering if a different method signature (e.g. `unfavorite(_:)`) might be a little clearer.

Reply:
I could see value in this change - I'll update it!